central/sensor/service/pipeline/clustermetrics/update_maxima.go

```diff
@@ -14,15 +14,40 @@ type BillingMetrics struct {
 }
 
 var (
+	// lastKnown stores the last known metrics per cluster.
+	lastKnown = map[string]BillingMetrics{}
+
+	// nodesMap and coresMap store the maximum numbers of nodes and cores
+	// per cluster.
 	nodesMap = maputil.NewMaxMap[string, int64]()
 	coresMap = maputil.NewMaxMap[string, int64]()
 )
 
 func updateMaxima(clusterID string, cm *central.ClusterMetrics) {
+	lastKnown[clusterID] = BillingMetrics{
+		TotalNodes: cm.GetNodeCount(),
+		TotalCores: cm.GetCpuCapacity(),
+	}
+
 	nodesMap.Add(clusterID, cm.GetNodeCount())
 	coresMap.Add(clusterID, cm.GetCpuCapacity())
 }
 
+// FilterCurrent removes the last known metrics values for the cluster IDs
+// not present in the ids, and returns the total values for other IDs.
+func FilterCurrent(ids set.StringSet) *BillingMetrics {
+	var m BillingMetrics
+	for id, v := range lastKnown {
+		if ids.Contains(id) {
+			m.TotalNodes += v.TotalNodes
+			m.TotalCores += v.TotalCores
+		} else {
+			delete(lastKnown, id)
```

Comment:
This also doesn't look concurrency safe to me since it's called from the service handler.

Reply:
My bad, I forgot to protect the map.